core/src/page_id.rs

```diff
@@ -155,12 +161,46 @@ impl PageId {
     /// Get the maximum descendant of this page.
     pub fn max_descendant(&self) -> PageId {
         let mut page_id = self.clone();
-        while page_id.path.len() < 42 {
+        while page_id.path.len() < MAX_PAGE_DEPTH {
             page_id.path.push(MAX_CHILD_INDEX);
         }
 
         page_id
     }
+
+    /// Get the minimum key-path which could land in this page.
+    pub fn min_key_path(&self) -> KeyPath {
+        let mut path = KeyPath::default();
+        for (i, child_index) in self.path.iter().enumerate() {
```

Comment:
maybe this could be a possible refactor to avoid `path.view_bits_mut()` on each iteration 
```rust
        path.view_bits_mut::<Msb0>()
            .chunks_exact_mut(6)
            .zip(self.path.iter())
            .for_each(|(chunk, child_index)| {
                chunk.copy_from_bitslice(&child_index.view_bits::<Msb0>()[2..8])
            });

        path.view_bits_mut::<Msb0>()[(6 * self.path.len())..256].fill(false);
```

Reply:
Thanks for the suggestion, but I find it more readable this way and it doesn't seem to need to be optimized.